native/zowed/worker.cpp

```diff
@@ -189,13 +450,315 @@ void WorkerPool::shutdown()
+    if (previous_ready <= 0)
+    {
+      // Critical: ready count and ready list are inconsistent
+      // Reset count to 0 and abort replacement to prevent further corruption
+      ready_count.store(0, std::memory_order_release);
+      LOG_ERROR("Ready count inconsistency detected for worker %zu (was %d). Aborting replacement to prevent corruption.",
+                worker_index, previous_ready);
+      return false;
+    }
+  }
+
+  return true;
+}
+
+struct ReplacementContext
+{
+  std::shared_ptr<Worker> old_worker;
+  bool max_attempts_reached;
+  size_t attempt_number;
+  std::chrono::milliseconds applied_backoff;
+
+  ReplacementContext()
+      : max_attempts_reached(false), attempt_number(0), applied_backoff(0)
+  {
+  }
+};
+
+ReplacementContext WorkerPool::prepare_worker_replacement(size_t worker_index)
+{
+  ReplacementContext ctx;
+  std::lock_guard<std::mutex> lock(ready_mutex);
+
+  if (worker_index >= workers.size())
+  {
+    LOG_ERROR("Cannot prepare worker replacement at index %zu - out of range", worker_index);
+    return ctx;
+  }
+
+  const size_t attempts = replacement_attempts[worker_index];
+  const auto now = std::chrono::steady_clock::now();
+
+  if (attempts >= max_replace_attempts)
+  {
+    ctx.max_attempts_reached = true;
+  }
+  else
+  {
+    // Not max attempts, so schedule the next backoff time
+    ctx.attempt_number = attempts + 1;
+    replacement_attempts[worker_index] = ctx.attempt_number;
+
+    auto computed_backoff = base_replace_backoff * (1ULL << (ctx.attempt_number - 1));
+    if (computed_backoff > max_replace_backoff)
+      computed_backoff = max_replace_backoff;
+
+    ctx.applied_backoff = computed_backoff;
+    next_replacement_allowed[worker_index] = now + computed_backoff;
+  }
+
+  // Move worker out to be processed outside the lock
+  ctx.old_worker = std::move(workers[worker_index]);
+  return ctx;
+}
+
+std::vector<RequestMetadata> WorkerPool::recover_requests_from_worker(
+    std::shared_ptr<Worker> &old_worker,
+    size_t worker_index,
+    const char *reason,
+    bool force_detach)
+{
+  std::vector<RequestMetadata> recovered_requests;
+
+  if (!old_worker)
+    return recovered_requests;
+
+  // Get pending requests from queue
+  auto pending = old_worker->drain_pending_requests();
+  recovered_requests.insert(recovered_requests.end(), pending.begin(), pending.end());
+
+  // Recover in-flight request only if it wasn't a timeout/hang
+  std::string current_req = old_worker->get_current_request();
+  if (!current_req.empty())
+  {
+    if (!force_detach)
+    {
+      LOG_DEBUG("Worker %zu: Recovering in-flight request due to %s", worker_index, reason);
+      recovered_requests.emplace_back(RequestMetadata(current_req, 0));
+    }
+    else
+    {
+      LOG_DEBUG("Worker %zu: Sending timeout error for in-flight request", worker_index);
+      RpcServer &server = RpcServer::get_instance();
+      server.send_timeout_error(current_req, request_timeout.count());
+    }
+  }
```

Comment:
yeah, this (https://github.com/zowe/zowe-native-proto/pull/651#discussion_r2490670581) was already addressed

Thank you! 🙏